### PR TITLE
fix: should reconcile targets consistently

### LINKED ISSUE #259: Bundle is continuously synced when PKCS12 is enabled
When the PKCS12 additional format is enabled trust-managed is continuously syncing the ConfigMaps and ultimately filling up etcd with old versions, compacting and defragging etcd results in it filling back up to several GB size within 15 minutes or so.

Nothing in the logs to suggest why (even with the logging level set to 5) and the logs indicate a successful sync but it fires a sync every seconds or so, stops as soon as I remove the PKCS12 format from the Bundle. Not happening with JKS.

K8s 1.27.7 - RKE2
trust-managed 0.7.0 - Installed from Helm

### PR DESCRIPTION
**Pull request by erikgb**
When checking target keys using managed fields, we must also check `binaryData` field in target configmaps.

Fixes #259

### CODE REVIEW

**Comment by inteon** [comment]
@erikgb just for context, do you know why the extra reconciliations and patches keep happening? It was expecting the noop patches to not trigger new reconciliations.

**Comment by erikgb** [comment]
@inteon Neither the JKS nor PKCS#12 format is idempotent, so when the check for keys never match it will patch the target with changes. I really don't know why this didn't happen with JKS only. The added tests only fail on PKCS#12 without the fix...

**Comment by inteon** [comment]
@erikgb I think we explicitly try to make the JKS encoded value deterministic: https://github.com/cert-manager/trust-manager/blob/54e8b096c2faf5e2f11454739d2abd4f5a691f75/pkg/bundle/sync.go#L191-L193

**Comment by inteon** [comment]
@erikgb It looks like the pkcs#12 format is using a random salt, which makes it non-idempotent.
see https://github.com/SSLMate/go-pkcs12/blob/v0.4.0/pkcs12.go#L704C5-L706

Now that I fully understand the cause of this issue and why it occurs for pkcs#12 & not for JKS, I'm more than happy to merge this.
/approve
/lgtm

File: test/integration/bundle/suite.go [hunk 1]
```diff
@@ -447,6 +447,41 @@ var _ = Describe("Integration", func() {
...
+			})()).To(Succeed())
+
+			configMap := &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Namespace: "kube-system", Name: testBundle.Name}}
+			Eventually(komega.Object(configMap)).Should(HaveField("BinaryData", HaveKey("target.jks")))
+
+			resourceVersion := configMap.ResourceVersion
+			Consistently(komega.Object(configMap)).Should(HaveField("ObjectMeta.ResourceVersion", Equal(resourceVersion)))
+		})
+
+		It("should have stable resourceVersion for PKCS12 target", func() {
```
Comment (inteon) [inline test/integration/bundle/suite.go:471]: I really like these tests, feel free to add these to other projects if you have some extra time 😄.

Comment (erikgb) [reply test/integration/bundle/suite.go:471]: > I really like these tests, feel free to also add these to other projects if you have some extra time 😄.

Thanks, I also like them. Reconcile-loop errors like this one can be very annoying, and I have been looking for a good way of testing/detecting them.... I'll see what I have time for. 😉

**Pull request merged**